moveit_ros/planning/kinematics_plugin_loader/src/kinematics_plugin_loader.cpp

```diff
@@ -369,52 +387,66 @@ robot_model::SolverAllocatorFn KinematicsPluginLoader::getLoaderFunction(const s
             }
             else
             {  // handle the case this param is just one value and parsed as a double
-              double res;
-              if (nh.getParam(ksolver_res_param_name, res))
-                search_res[known_groups[i].name_].push_back(res);
-              else
+              if (ksolver_params->has_parameter(ksolver_res_param_name))
```

Comment:
This condition is always false since we are in the `else` case of the upper condition.

Reply:
We should probably break up this if/else condition and just query the parameter once. After that we can handle the different cases (string/int/double). Ideally, we should be able to just cast the int to double and only use one line.